utils/s2n_mem.c

```diff
@@ -46,7 +46,12 @@ static s2n_mem_free_callback s2n_mem_free_cb = s2n_mem_free_mlock_impl;
 
 static int s2n_mem_init_impl(void)
 {
+/* Avoids spurious alarms over signed to unsigned conversion in (uint32_t)return_value_sysconf. */
+#pragma CPROVER check push
+#pragma CPROVER check disable "conversion"
     GUARD(page_size = sysconf(_SC_PAGESIZE));
```

Comment:
Is there a reason we don't just use the same type for both?

Reply:
`page_size` is a global variable and we can't change the return type for sysconf, unless we use a stub for this proof. Do you think the stub would be better than the pragma?

Reply:
Actually, if this can return -1, we should store the result to a long, check it for -1, check that it fits within a uint32 otherwise, and then store it.

Reply:
oh the joys of platform-variant sized integers...

Reply:
Implemented as suggested.